Sources/AWSLambdaEvents/APIGateway+WebSockets.swift

```diff
@@ -0,0 +1,68 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) YEARS Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// `APIGatewayWebSocketRequest` is a variation of the`APIGatewayV2Request`
+/// and contains data coming from the WebSockets API Gateway.
+public struct APIGatewayWebSocketRequest: Codable {
```

Comment:
can any of this vibe shared with `APIGatewayRequest` or not worth it?

Reply:
Oh sure ... much of it can be. I just wasn't sure what the correct approach should be. I kind of aimed for a "what's the minimum to make it work" approach. :)

Reply:
probably some shared struct they can both include as the underlying implementation

Reply:
If it would be helpful, I would be happy to look at the `APIGatewayV2Request` event, as well as my proposed `APIGatewayWebSocketRequest` event and extract those items shared between them as a `struct` both can use…and then update this pull request.

Reply:
If it would be helpful, I would be happy to look at the `APIGatewayV2Request` event, as well as my proposed `APIGatewayWebSocketRequest` event and extract those items shared between them as a `struct` both can use…and then update this pull request.